deploy/olm-catalog/smart-gateway-operator/manifests/smartgateway.infra.watch_smartgateways_crd.yaml

```diff
@@ -34,75 +34,130 @@ spec:
-                format
+              description: Registry path to the sg-core container image.
               type: string
-            dataCount:
-              description: Stop after receiving this many messages in total (0 forever).
-                Defaults to '0'.
-              type: integer
-            debug:
-              description: Enable verbose debugging statements. Defaults to 'false'.
+            handleErrors:
+              description: Tell handlers to send errors to event bus.
               type: boolean
-            elasticPass:
-              description: Basic Authentication password for ElasticSearch.
-              type: string
-            elasticUrl:
-              description: URL for ElasticSearch endpoing to connect when the Smart
-                Gateway is operating as an 'events' service type. Defaults to 'elasticsearch-es-http.<namespace>.svc:9200'.
-              type: string
-            elasticUser:
-              description: Basic Authentication username for ElasticSearch.
+            logLevel:
+              description: Log level for sg-core.
               type: string
-            exporterHost:
-              description: Metrics URL for Prometheus to export. Defaults to "0.0.0.0".
-              type: string
-            exporterPort:
-              description: Metrics port for Prometheus to export. Defaults to 8081.
-              type: integer
-            resetIndex:
-              description: Reset the ElasticSearch index on load. Defaults to 'false'.
-              type: boolean
-            serviceType:
-              description: The service type for this smart gateway. One of 'metrics'
-                or 'events'. Defaults to 'metrics'.
-              enum:
-              - metrics
-              - events
+            pluginDir:
+              description: Plugin directory path.
               type: string
+            services:
+              description: Services that should be created for this Smart Gateway
+              items:
+                properties:
+                  name:
+                    description: Postfix of the service name
+                    type: string
+                  ports:
+                    description: Ports to create for this service
+                    items:
+                      properties:
+                        name:
+                          description: Port name
+                          type: string
+                        port:
+                          description: Port number to expose
+                          type: integer
+                        protocol:
+                          description: Protocol for this service
+                          type: string
+                        targetPort:
+                          description: Target port for service
+                          type: integer
+                    type: array
+              type: array
```

Comment:
Another example where not inferring the services to expose from an application configuration makes the logic in SGO simpler. Just define the list of services you need your application to expose and you're done. Will allow different services to be exposed in the future for applications we're not using yet.

Reply:
I'm not 100% sold on this approach since it seems like more work for the user. In practice I doubt this will really be an issue, and it's definitely easier to maintain this way, so okay!

Reply:
From the user perspective in 99% of the cases (or more) this will be totally transparent as the user interface will be Service Telemetry Operator.

I'm mostly thinking about the additional plugins we're going to be using, and I'm trying to reduce the need to adjust the SGO when an STO side change is required.